### PR TITLE
Add fileStorageUploadField and fileUploadField Components to UI Samples

### PR DESCRIPTION
**Pull request by AmirShelli**
This pull request introduces two new components to the UI samples: `fileStorageUploadField` and `fileUploadField`.
Added their description and its Russian localization.

### CODE REVIEW

File: src/main/java/io/jmix/uisamples/entity/Document.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+import io.jmix.core.FileRef;
+import io.jmix.core.entity.annotation.JmixGeneratedValue;
+import io.jmix.core.entity.annotation.JmixId;
+import io.jmix.core.metamodel.annotation.JmixEntity;
+import io.jmix.core.metamodel.annotation.JmixProperty;
+
+import java.util.UUID;
+
+@JmixEntity
```
Comment (honest-niceman) [inline src/main/java/io/jmix/uisamples/entity/Document.java:11]: Please look other entities declaration. @Entity and @Table annotations are missing here

File: src/main/java/io/jmix/uisamples/view/flowui/components/fileupload/filestorageuploadfield/FileStorageUploadFieldSample.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+import java.io.File;
+import java.util.UUID;
+
+@ViewController("file-storage-upload-field")
+@ViewDescriptor("file-storage-upload-field.xml")
+public class FileStorageUploadFieldSample extends StandardView {
+    @Autowired
+    private TemporaryStorage temporaryStorage;
+    @ViewComponent
```
Comment (honest-niceman) [inline src/main/java/io/jmix/uisamples/view/flowui/components/fileupload/filestorageuploadfield/FileStorageUploadFieldSample.java:22]: Do not mix @Autowired and @ViewComponent fields. Please define all autowired field first, then all @ViewComponent

File: src/main/java/io/jmix/uisamples/view/flowui/components/fileupload/filestorageuploadfield/FileStorageUploadFieldSample.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    private FileStorageUploadField fileStorageUploadField;
+    @Autowired
+    private Notifications notifications;
+
+    @Subscribe("fileStorageUploadField")
+    public void onFileStorageUploadFieldFileUploadSucceeded(
+            final FileUploadSucceededEvent<FileStorageUploadField> event) {
+        Receiver receiver = event.getReceiver();
+        if (receiver instanceof FileTemporaryStorageBuffer) {
```
Comment (honest-niceman) [inline src/main/java/io/jmix/uisamples/view/flowui/components/fileupload/filestorageuploadfield/FileStorageUploadFieldSample.java:31]: Use smart instanceof construction from java17. E.g:

if (receiver instanceof FileTemporaryStorageBuffer storageBuffer) {

File: src/main/java/io/jmix/uisamples/view/flowui/components/fileupload/filestorageuploadfield/FileStorageUploadFieldSample.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+import io.jmix.flowui.view.*;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.io.File;
+import java.util.UUID;
+
+@ViewController("file-storage-upload-field")
+@ViewDescriptor("file-storage-upload-field.xml")
+public class FileStorageUploadFieldSample extends StandardView {
+    @Autowired
```
Comment (honest-niceman) [inline src/main/java/io/jmix/uisamples/view/flowui/components/fileupload/filestorageuploadfield/FileStorageUploadFieldSample.java:20]: Add empty line before first field

File: src/main/resources/io/jmix/uisamples/view/flowui/components/avatar/avatar-en.html [hunk 5]
```diff
@@ -1,4 +1,4 @@
-<p>A avatar can contain a name, an abbreviation, an icon. By default, the abbreviation is the first characters of the name.</p>
+<p>An avatar can contain a name, an abbreviation, an icon. By default, the abbreviation is the first characters of the name.</p>
```
Comment (honest-niceman) [inline src/main/resources/io/jmix/uisamples/view/flowui/components/avatar/avatar-en.html:1]: Basically, this change has nothing related to this PR. Isn't it?

File: src/main/resources/io/jmix/uisamples/menu/uisamples-menu.xml [hunk 6]
```diff
@@ -20,6 +20,18 @@
     <menu id="components-root" url="flow-ui/vc/components">
         <about id="components" location="components/about/about.html"/>
         <item id="avatar" page="avatar"/>
+        <menu id="upload-field">
```
Comment (honest-niceman) [inline src/main/resources/io/jmix/uisamples/menu/uisamples-menu.xml:23]: The component should be added in alphabetic order, not as the second item

File: src/main/resources/io/jmix/uisamples/menu/uisamples-menu.xml [hunk 7]
```diff
@@ -20,6 +20,18 @@
...
+            <item id="file-upload-field" page="file-upload-field">
+                <otherFiles>
+                    <file name="io/jmix/uisamples/entity/Picture.java"/>
```
Comment (honest-niceman) [inline src/main/resources/io/jmix/uisamples/menu/uisamples-menu.xml:26]: Check out the documentation link. It leads to the 404 page

File: src/main/java/io/jmix/uisamples/view/flowui/components/fileupload/fileuploadfield/FileUploadFieldSample.java [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+
+    @Subscribe("fileUploadField")
+    public void onFileUploadFieldFileUploadSucceeded(
+            final FileUploadSucceededEvent<FileStorageUploadField> event) {
+        String fileName = event.getFileName();
+        notifications.create("Your file \""
+                        + fileName + "\" has been uploaded successfully.").withThemeVariant(NotificationVariant.LUMO_PRIMARY)
+                .show();
+    }
+}
```
Comment (honest-niceman) [inline src/main/java/io/jmix/uisamples/view/flowui/components/fileupload/fileuploadfield/FileUploadFieldSample.java:28]: Keep looking at file ends. Only one empty line at the end. Not two or zero empty lines.

File: src/main/java/io/jmix/uisamples/view/flowui/components/fileupload/filestorageuploadfield/FileStorageUploadFieldSample.java [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+    private TemporaryStorage temporaryStorage;
+    @ViewComponent
+    private FileStorageUploadField fileStorageUploadField;
+    @Autowired
+    private Notifications notifications;
+
+    @Subscribe("fileStorageUploadField")
+    public void onFileStorageUploadFieldFileUploadSucceeded(
+            final FileUploadSucceededEvent<FileStorageUploadField> event) {
```
Comment (honest-niceman) [inline src/main/java/io/jmix/uisamples/view/flowui/components/fileupload/filestorageuploadfield/FileStorageUploadFieldSample.java:29]: Do we have a linter for exceeded line length? If not, i think it's better to not split method declaration into two lines

File: src/main/resources/io/jmix/uisamples/messages_en.properties [hunk 10]
```diff
@@ -165,6 +172,10 @@ uisamples-menu-config.components=Components overview
 
 uisamples-menu-config.avatar = Avatar
 
+uisamples-menu-config.upload-field = UploadField
```
Comment (honest-niceman) [inline src/main/resources/io/jmix/uisamples/messages_en.properties:175]: I think it's better to name it "Managing files"

**Review by honest-niceman — CHANGES_REQUESTED** [review]
Please check out ny suggestions

**Comment by AmirShelli** [comment]
Implemented the requested changes.

**Review by honest-niceman — APPROVED** [review]
Looks good to me

**Comment by honest-niceman** [comment]
@knstvk please take a look, LGTM

**Comment by knstvk** [comment]
@AmirShelli - please explain in the description:
- how to use the example: what file types and sizes are accepted.
- the fact that the example of fileStorageUploadField does not actually saves the file in the storage. Add a comment to the code explaining that it's just a demo code and should not be used in production.

Also, how the drop zone is supposed to work? When I drag-drop a file to the component from macOS Finder, the file is downloaded instead of uploaded.

**Comment by AmirShelli** [comment]
@knstvk 
Implemented the requested changes:
* Adjusted the description accordingly
* Included instructions on how to use the drag-and-drop zone
* Changed the section name to "File Upload Fields"

**Pull request merged**